src/reducers/select.js

```diff
@@ -0,0 +1,197 @@
+import {normalizeSync} from 'normalize-diacritics';
+
+export default (pattern) => (base, source) => {
+  const debug = createDebugLogger('@natlibfi/marc-record-merge');
+  const baseFieldsFromPattern = base.get(pattern);
+  const sourceFieldsFromPattern = source.get(pattern);
+  return selectFields();
+
+  function selectFields() {
+    // Check that all fields are data fields
+    // Test 01: The field is a control field (contains 'value')
+    const baseFields = checkFieldType(baseFieldsFromPattern);
+    //debug(`baseFields: ${JSON.stringify(baseFields, undefined, 2)}`);
+    const sourceFields = checkFieldType(sourceFieldsFromPattern);
+    //debug(`sourceFields: ${JSON.stringify(sourceFields, undefined, 2)}`);
+      
+    // Test 02: If there are multiple fields, the base field is returned unmodified
+    if (baseFields.length > 1 || sourceFields.length > 1) {
+      debug(`Multiple fields in base or source`);
+      return base;
+    }
+
+    // Check that the base and source tags are equal
+    // Test 03: Base and source tags are equal for one field (continue to comparison of contents)
+    // Test 04: Base and source tags are not equal (base field returned unmodified)
+    // Note: to check this, the pattern has to allow two tags, otherwise the unequal tag does not even pass source.get(pattern)
+    if (checkTags(baseFields, sourceFields) === false) {
+      debug(`Tags are not equal`);
+      return base;
+    }
+    
+    // Test 05: Normalize subfields of both base and source fields in order:
+    // 1. Remove diacritics
+    // 2. Change to lowercase
+    // 3. Trim whitespace at ends and replace sequential whitespace with a single whitespace character
+    // 4. Remove punctuation characters
+    baseFields.forEach(field => field.subfields.forEach(normalizeDiacritics));
+    baseFields.forEach(field => field.subfields.forEach(changeToLowerCase));
+    baseFields.forEach(field => field.subfields.forEach(removeWhitespace));
+    baseFields.forEach(field => field.subfields.forEach(removePunctuation));
+    //debug(`baseFields normalized: ${JSON.stringify(baseFields, undefined, 2)}`);
+    sourceFields.forEach(field => field.subfields.forEach(normalizeDiacritics));
+    sourceFields.forEach(field => field.subfields.forEach(changeToLowerCase));
+    sourceFields.forEach(field => field.subfields.forEach(removeWhitespace));
+    sourceFields.forEach(field => field.subfields.forEach(removePunctuation));
+    //debug(`sourceFields normalized: ${JSON.stringify(sourceFields, undefined, 2)}`);
+
+    // Compare equality of normalized subfields:
+    // 1. Default: Strict equality (subfield codes and values are equal)
+    // 2. Subset comparison (equal codes, one subfield value may be a subset of the other)
+    // Test 06: Two subfields, both equal
+    // Test 07: Two subfields, same codes, values are mutual subsets
+    // Test 08: Two subfields, one equal, one not
+    // Test 09: Two subfields, same codes, values are not subsets
+    
```

Comment:
Implement this: `Calculate the difference of subfields using the equality comparison function: How many equal subfields are there in source for base and vice versa.`